pkg/pattern/drain/chunk.go

```diff
@@ -43,9 +43,9 @@ func (c Chunk) spaceFor(ts model.Time) bool {
 }
 
 // ForRange returns samples with only the values
-// in the given range [start:end).
-// start and end are in milliseconds since epoch.
-func (c Chunk) ForRange(start, end model.Time) []logproto.PatternSample {
+// in the given range [start:end) and aggregates them by step duration.
+// start and end are in milliseconds since epoch. step is a duration in milliseconds.
+func (c Chunk) ForRange(start, end, step model.Time) []logproto.PatternSample {
```

Comment:
I'm not 100% sure this is the best place to add this functionality. It could easily be separated into a separate "groupByStep" function or similar.

Reply:
I think it's fine for now, the only problem I see is if someone uses multiple chunks with different start,end,step. Then it going to be a mess, but right now that's not the case so happy with this direction.